asimov-deploy/asimov-deploy
# Various fixes for autopilot/automatic deploy, ui tests

First of all, sorry for the huge pull request!!

Automatic deploy wizard:

- Added possibility to exclude deployable unit(s) with checkboxes on the second step

Automatic deploy:

- Added retry logic to deploy task if a deploy fails. This is configurable on a global level for now, see config.json below.
- Added retry logic to verify task if a verify step fails. This is configurable on a global level for now, see config.json below.

Enable UI tests using karma, jasmine and phantomjs:

- Tests of autopilot tasks
- Provide dependencies to tasks thru constructor to ease unit testing
- Autopilot/automatic deploy now has some default settings for tasks that can be configured thru config.json

```json
{
   ...
   "autopilot": {
      ...
        "settings": {
            "loadBalancerTimeout": 10000,
            "enableLoadBalancerPostDelay": 1000,
            "disableLoadBalancerPostDelay": 500,
            "deployRetryOnFailure": true,
            "deployFailureRetries": 2,
            "verifyPostDelay": 1000,
            "verifyRetryOnFailure": true,
            "verifyFailureRetries": 2
        }
    }
}
```

Demo-mode changes:

- Use group filtering feature in demo-mode
- Load balancer should not be able to have a negative connection count in demo-mode

Other:

- Added license field in package.json to get rid of npm warning
